internal/integratedservices/cleanup.go

```diff
@@ -0,0 +1,122 @@
+// Copyright © 2021 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integratedservices
+
+import (
+	"context"
+	"time"
+
+	"emperror.dev/errors"
+	"github.com/banzaicloud/integrated-service-sdk/api/v1alpha1"
+	"github.com/banzaicloud/operator-tools/pkg/utils"
+	apiErrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type integratedServiceClean struct {
+	scheme       *runtime.Scheme
+	kubeConfigFn ClusterKubeConfigFunc
+}
+
+func NewIntegratedServiceClean(kubeConfigFn ClusterKubeConfigFunc) IntegratedSeriviceCleaner {
+	scheme := runtime.NewScheme()
+	_ = clientgoscheme.AddToScheme(scheme)
+	_ = v1alpha1.AddToScheme(scheme)
+
+	return integratedServiceClean{
+		scheme:       scheme,
+		kubeConfigFn: kubeConfigFn,
+	}
+}
+
+func (r integratedServiceClean) DisableServiceInstance(ctx context.Context, clusterID uint) error {
+	clusterClient, err := r.k8sClientForCluster(ctx, clusterID)
+	if err != nil {
+		return errors.Wrap(err, "failed to build cluster client")
+	}
+
+	lookupISvcs := &v1alpha1.ServiceInstanceList{}
+	if err := clusterClient.List(ctx, lookupISvcs); err != nil {
+		if meta.IsNoMatchError(err) {
+			return nil
+		}
+		return errors.Wrap(err, "failed to retrieve service instance list")
+	}
+
+	for _, item := range lookupISvcs.Items {
+		if item.ObjectMeta.Annotations["app.kubernetes.io/managed-by"] == "banzaicloud.io/pipeline" && item.Spec.Enabled != nil &&
+			*item.Spec.Enabled == true {
```

Comment:
maybe I suggested this, but when cleaning the cluster we should disable all services not just the ones managed by pipeline

```suggestion
		if utils.PointerToBool(item.Spec.Enabled) {
```